src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java

```diff
@@ -75,7 +75,7 @@ public boolean equals(Object another) {
      * @return the contents of this principal in the form realm:username
      */
     public String getName() {
-        return username;
+        return String.format("%s:%s", username, realm);
```

Comment:
Isn't it the opposite you should do Patrick :-) ?
```realm:username```

Reply:
Fixed in commit 2ad0c8d8a0d21713b6b3cdbd59306fd4d57bf4d7